### PR TITLE
Added .idea to gitignore and updated checkstyle

### PR DESCRIPTION
**Pull request by freddd**
Added the NeedBraces check. Once this is merged, the following code block is no
longer permitted.
```
if (false)
 doSomething();
```
It needs to have braces:
```
if (false) {
    doSomething();
}
```

Ping @adamvoncorswant, do I need to change versions anywhere else or will this be done automatically when we do mvn release:perform? We of course need to change it in all of the other projects but that we'll do one at a time to fix what we break.

### CODE REVIEW

File: checkstyle/src/main/resources/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -17,6 +17,9 @@
             <property name="arrayInitIndent" value="4"/>
             <property name="lineWrappingIndentation" value="4"/>
         </module>
+        <module name="NeedBraces">
+            <property name="tokens" value="LITERAL_DO, LITERAL_ELSE, LITERAL_IF, LITERAL_FOR, LITERAL_WHILE"/>
```
Comment (adamvoncorswant) [inline checkstyle/src/main/resources/checkstyle/checkstyle.xml]: All these tokens are the default. Don't you think we should do with an empty tag instead? Like so:
```xml
<module name="NeedBraces"/>
```

File: basepom/pom.xml [hunk 2]
```diff
@@ -268,7 +268,7 @@
                         <dependency>
                             <groupId>com.izettle</groupId>
                             <artifactId>izettle-checkstyle</artifactId>
-                            <version>2.1.11</version>
+                            <version>2.1.12</version>
```
Comment (adamvoncorswant) [inline basepom/pom.xml]: I must admin that I still don't understand the setup for this specific repo. Let's verify with @nzroller tomorrow :)

**Comment by adamvoncorswant** [comment]
Love this BTW. I cannot understand why it wasn't included in our previous configuration...

**Comment by nzroller** [comment]
@adamvoncorswant the two projects are independent but closely related, so I haven't been using `mvn release:prepare/perform` at all. That may have been because there was a release plugin bug making releases in projects that weren't at the root of the git repo problematic. Good news is that I think that has been fixed so @freddd and I can try it out and we'll see what happens though and we'll add to the readme or make it easier for future changes. @freddd tell me when you want to do it!

**Comment by freddd** [comment]
Released & all done.

**Comment by adamvoncorswant** [comment]
Nice. Thanks guys!

**Comment by nzroller** [comment]
@freddd I reckon we should add the ability to disable this (`NeedsBraces`) around `equals` and `hashcode` methods. I don't think it's that beneficial to have to go around and change these IDE generated methods. Both intellij and eclipse generate these methods without braces. What do you think? We could enable the [supressionCommentFilter](http://stackoverflow.com/questions/4023185/how-to-disable-a-particular-checkstyle-rule-for-a-particular-line-of-code/4023351#4023351)?

**Comment by freddd** [comment]
Yes, I agree, that would be good.

**Pull request merged**